src/libs/actions/User.js

```diff
@@ -58,9 +61,35 @@ function changePassword(oldPassword, password) {
 }
 
 function getBetas() {
-    API.User_GetBetas().then((response) => {
+    return API.User_GetBetas().then((response) => {
         if (response.jsonCode === 200) {
             Onyx.set(ONYXKEYS.BETAS, response.betas);
+            return response.betas;
+        }
+        return [];
+    });
+}
+
+/**
+ * Fetches betas and User's preferred Locale, then set the app locale.
+ * @param {String} currentPreferredLocale
+ */
+function fetchBetasAndSetPreferredLocale(currentPreferredLocale) {
+    Promise.all([
+        getBetas(),
+        API.Get({
+            returnValueList: 'nameValuePairs',
+            nvpNames: ONYXKEYS.NVP_PREFERRED_LOCALE,
+        }),
+    ]).then((betas, response) => {
+        const preferredLocale = lodashGet(response, ['nameValuePairs', 'preferredLocale'], CONST.DEFAULT_LOCALE);
+        if (currentPreferredLocale !== CONST.DEFAULT_LOCALE
+            && preferredLocale !== currentPreferredLocale
+            && Permissions.canUseInternationalization(betas)
+        ) {
+            setLocale(currentPreferredLocale);
+        } else {
+            Onyx.set(ONYXKEYS.NVP_PREFERRED_LOCALE, preferredLocale);
```

Comment:
What's the difference between `setLocale()` and `Onyx.set(ONYXKEYS.NVP_PREFERRED_LOCALE, value)` 🤔  ?